### PR TITLE
add suppression policy at state level

### PR DESCRIPTION
**Pull request by zhang-xinyu**
This PR addresses issue https://github.com/covid-projections/covid-data-model/issues/< >

Add function to interpolate state level suppression policy, and update other functions as follow:
- add parameter `reference_start_date` as it is needed to align interpolation of policy of different counties. 
- add function to infer t0 and return default value '2020-01-15' if there is no imputed t0 or observed case.

Add function to run state level MLE inference to `model_fitter_mle.py` 

### Please Check
- [ ] Will the *current* schema in [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.md) be updated with this PR?
- [ ] Are tests passing?

### CODE REVIEW

File: pyseir/inference/model_fitter_mle.py [hunk 1]
```diff
@@ -156,12 +283,91 @@ def plot_inferred_result(fit_results):
...
+    Plot the results of an MLE inference at state level.
+    """
+    state = fit_results['state']
+    metadata = load_data.load_county_metadata_by_state(state) \
+                        .loc[state] \
+                        .to_dict()
+    times, observed_new_cases, observed_new_deaths = \
+            load_data.load_new_case_data_by_state(state, ref_date)
+
+    if observed_new_cases.sum() < 5:
```
Comment (erccarls) [inline pyseir/inference/model_fitter_mle.py:334]: parameterize the 5

File: pyseir/inference/model_fitter_mle.py [hunk 2]
```diff
@@ -183,11 +390,18 @@ def run_state(state, states_only=False):
         output_file = os.path.join(OUTPUT_DIR, state.title(), 'data', f'summary_{state}__mle_fit_results.json')
         pd.DataFrame(fit_results).to_json(output_file)
 
-        p.map(plot_inferred_result, fit_results)
+        p.map(plot_inferred_result_county, fit_results)
         p.close()
 
+    else:
+        fit_results = fit_state_model(state)
+        output_file = os.path.join(OUTPUT_DIR, state.title(), 'data',
```
Comment (erccarls) [inline pyseir/inference/model_fitter_mle.py:390]:
```suggestion
        output_file = os.path.join(OUTPUT_DIR, 'pyseir', state.title(), 'data',
```

File: pyseir/load_data.py [hunk 3]
```diff
@@ -250,6 +291,44 @@ def load_new_case_data_by_fips(fips, t0):
...
+    _state_fips = load_county_metadata()[['state', 'fips']]
+    _state_fips = _state_fips[_state_fips.state == state].drop_duplicates()
+
+    # set long enough lists to record new cases and deaths count
+    observed_new_cases_state = np.zeros(360)
+    observed_new_deaths_state = np.zeros(360)
+    times_new_state = np.zeros(360)
+    for fips in _state_fips['fips']:
+        times, observed_new_cases, observed_new_deaths = \
+            load_new_case_data_by_fips(fips, t0)
```
Comment (erccarls) [inline pyseir/load_data.py:323]: I think we can swap this to their data loaders which perform the aggregation.

```
from libs.datasets import FIPSPopulation, JHUDataset, CDSDataset
from libs.datasets.dataset_utils import build_aggregate_county_data_frame
from libs.datasets.dataset_utils import AggregationLevel

jhu_local = JHUDataset.local()

df = jhu_local.timeseries() \
    .get_subset(AggregationLevel.STATE, country='USA') \
    .get_data(country='USA', state='CA')
```

File: pyseir/models/suppression_policies.py [hunk 4]
```diff
@@ -155,7 +157,41 @@ def get_interventions(start_date=datetime.now().date()):
     return interp1d(t_list, rho, fill_value='extrapolate')
 
 
-def generate_empirical_distancing_policy(t_list, fips, future_suppression):
+def infer_t0(fips, method='first_case', default=pd.Timestamp('2020-01-15')):
```
Comment (erccarls) [inline pyseir/models/suppression_policies.py:160]: This should go in the inference folder

File: pyseir/models/suppression_policies.py [hunk 5]
```diff
@@ -228,7 +268,64 @@ def generate_empirical_distancing_policy(t_list, fips, future_suppression):
...
+    suppression_model: callable
+        suppression_model(t) returns the current suppression model at time t
+    """
+    
+    county_metadata = load_county_metadata()
+    counties_fips = county_metadata[county_metadata.state == state].fips.unique()
+    
+    if reference_start_date is None:
+        t0s = list()
+        for fips in counties_fips:
```
Comment (erccarls) [inline pyseir/models/suppression_policies.py:304]: `min([infer_t0(fips) for fips in counties_fips])`

File: pyseir/inference/model_fitter_mle.py [hunk 6]
```diff
@@ -33,6 +34,81 @@ def get_average_SEIR_parameters(fips):
...
+        suppression_policy = \
+        suppression_policies.generate_empirical_distancing_policy_by_state(
+            state=state, future_suppression=eps, **suppression_policy_params)
+
+    model = SEIRModel(
+        R0=R0, suppression_policy=suppression_policy,
+        **SEIR_params)
+    model.run()
+
+    predicted_cases = model.gamma * np.interp(times, t_list + t0, model.results[
```
[the remark this answers is not in the document]
Comment (zhang-xinyu) [reply pyseir/inference/model_fitter_mle.py:85]: Agreed, or we could have 'tested' time series in results and fit it to the observation.

File: pyseir/inference/model_fitter_mle.py [hunk 7]
```diff
@@ -107,47 +165,116 @@ def _fit_seir(R0, t0, eps):
     values['state'] = county_metadata['state']
     values['total_population'] = county_metadata['total_population']
     values['population_density'] = county_metadata['population_density']
+    values['I_initial'] = SEIR_params['I_initial']
```
[the remark this answers is not in the document]
Comment (zhang-xinyu) [reply pyseir/inference/model_fitter_mle.py:162]: They are needed for MLE trajectory plot, especially for state level. To make keys consistent, I also add it for county fit results.

File: pyseir/inference/model_fitter_mle.py [hunk 8]
```diff
@@ -13,7 +14,7 @@
 
 t_list = np.linspace(0, 1000, 1001)
 ref_date = datetime(year=2020, month=1, day=1)
-
+frac_counties_with_seed_infection = 0.5
```
[the remark this answers is not in the document]
Comment (zhang-xinyu) [reply pyseir/inference/model_fitter_mle.py:17]: Fraction of counties with seed infection when making state level inference. One seed infection is too few for state level which results in very few predicted cases/deaths.

File: pyseir/inference/model_fitter_mle.py [hunk 9]
```diff
@@ -33,6 +34,81 @@ def get_average_SEIR_parameters(fips):
...
+
+    model = SEIRModel(
+        R0=R0, suppression_policy=suppression_policy,
+        **SEIR_params)
+    model.run()
+
+    predicted_cases = model.gamma * np.interp(times, t_list + t0, model.results[
+        'total_new_infections'])
+    predicted_deaths = np.interp(times, t_list + t0, model.results[
+        'direct_deaths_per_day'])
```
[the remark this answers is not in the document]
Comment (zhang-xinyu) [reply pyseir/inference/model_fitter_mle.py:88]: Do you mean 'total_deaths_per_day'?

File: pyseir/inference/model_fitter_mle.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (erccarls) [reply pyseir/inference/model_fitter_mle.py:97]: Yeah. Only hard thing there is that it adds shape parameters that will likely make inference difficult (i.e. it looks about the same as changing the suppression policy).

File: pyseir/inference/model_fitter_mle.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (erccarls) [reply pyseir/inference/model_fitter_mle.py]: Yes. I need to clean this up soon. Everything is pretty messy right now on my side.

**Pull request merged**